### PR TITLE
library - get role

### PR DESCRIPTION
**Pull request by fabriziopandini**
This PR adds a method to the `Node` object for reading the `io.k8s.sigs.kind.role` label.

This method is really useful (or I would rather say it is necessary) when using kind as a library on an existing cluster, and you must use the list of actual nodes as a source of truth for your cluster topology e.g.
```go
nodes, _ := c.ListNodes()

for _, n := range nodes {
    if n.Role() == config.ControlPlaneRole {
        // Do something on roles
    }
}
```

### CODE REVIEW

File: pkg/cluster/nodes/node.go [hunk 1]
```diff
@@ -259,6 +263,24 @@ func (n *Node) Ports(containerPort int) (hostPort int, err error) {
...
+	if n.nodeCache.role != "" {
+		return n.nodeCache.role, nil
+	}
+	// retrive the role the node using docker inspect
+	lines, err := docker.Inspect(n.nameOrID, fmt.Sprintf("{{index .Config.Labels %q}}", constants.ClusterRoleKey))
+	if err != nil {
+		return "", errors.Wrapf(err, "failed to get %q label", constants.ClusterRoleKey)
+	}
+	if len(lines) != 1 {
+		return "", fmt.Errorf("%q label should only be one line, got %d lines", constants.ClusterRoleKey, len(lines))
```
Comment (neolit123) [inline pkg/cluster/nodes/node.go]: `errors.Errorf`.

Comment (fabriziopandini) [reply pkg/cluster/nodes/node.go]: done

File: pkg/cluster/nodes/node.go [hunk 2]
```diff
@@ -259,6 +263,24 @@ func (n *Node) Ports(containerPort int) (hostPort int, err error) {
...
 }
 
+// Role returns the role of the node
+func (n *Node) Role() (role config.NodeRole, err error) {
+	// use the cached version first
+	if n.nodeCache.role != "" {
+		return n.nodeCache.role, nil
+	}
+	// retrive the role the node using docker inspect
+	lines, err := docker.Inspect(n.nameOrID, fmt.Sprintf("{{index .Config.Labels %q}}", constants.ClusterRoleKey))
```
Comment (BenTheElder) [inline pkg/cluster/nodes/node.go]: We should probably move logic to obtain labels into the docker package at some point. This is consistent for now though and this PR is nice and small 👍

Comment (fabriziopandini) [reply pkg/cluster/nodes/node.go]: :+1:

**Pull request merged**